eventuals/do-all.h

```diff
@@ -57,7 +74,22 @@ struct _DoAll {
                               std::forward<decltype(errors)>(errors)...));
                   if (counter_.fetch_sub(1) == 1) {
                     // You're the last eventual so call the continuation.
-                    k_.Start(std::move(values_));
+                    std::optional<std::exception_ptr> exception =
+                        GetExceptionIfExists();
+
+                    if (exception) {
+                      try {
+                        std::rethrow_exception(*exception);
+                      } catch (const StoppedException&) {
+                        k_.Stop();
+                      } catch (...) {
+                        k_.Fail(std::current_exception());
+                      }
+                    } else {
+                      k_.Start(GetTupleOfValues());
+                    }
```

Comment:
At this point we should know that at least one exception exists because we just added one earlier in the lambda.
```suggestion
                    CHECK(exception);
                    try {
                      std::rethrow_exception(*exception);
                    } catch (const StoppedException&) {
                      k_.Stop();
                    } catch (...) {
                      k_.Fail(std::current_exception());
                    }
```